### PR TITLE
Setup friendships

### PR DESCRIPTION
**Pull request by luisvinicius09**
### What you can do:
- `Create` a friendship with another user
- `Accept` a friendship.
- `Refuse` a friendship

> Still improving the code, be gentle with it.

### CODE REVIEW

**Comment by luisvinicius09** [comment]
## Hellooo dear code reviewer.... I hope you are doing great.
### This comment is the documentation of already known problems to facilitate your life and of course, remind me to fix it.
##
### Starting for the easy to correct:
- [ ] `Friends` page is only displaying friends if there are friends.
- [ ] `Friendship requests` page is only displaying request if there is any.
- [ ] There is a lot of logic in the views, I'm working on removing them!

### Errors because I'm dumb:
- [x] The friendship request page is displaying the requests the user sent instead of received ones(still thinking how to solve it)

### Now, problems that I don't understand why are happening:
- [x] _~~Rspec is throwing the following error, but the cause is undefined for me. (still searching for it)~~_ **(FIXED)**
image

> Still working on the project! New updates soon.

File: app/views/friendships/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  <div class="request">
+    <h3 class='user-name'><%= fs.friend.name %></h3>
+
+    <%= form_with(model: @friendship, url: 'friendships/'+ fs.id.to_s, method: 'PATCH', local: true) do |form| %>
+      <%= form.hidden_field :friend_id, value: fs.friend_id %>
+        <div class="btn-submit">
+          <%= form.submit 'Accept'%>
+        </div>
+      <% end %>
+
```
Comment (acushlakoncept) [inline app/views/friendships/index.html.erb]: There should be absolutely NO logic in views.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -19,6 +19,7 @@
         </div>
       </div>
       <% if current_user %>
+        <%= link_to current_user.name, user_path(current_user) %>
         <%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
       <% else %>
         <%= link_to 'Sign in', user_session_path %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb:25]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @luisvinicius09,  Great job! 👏  Your progress is commendable 💪 

But there are still some changes that are necessary for this project to be approved.

- [x] There should be a button to invite user to friendship on users list page and on single user page, the same should be for Accepting or Rejecting Request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] Posts timeline should include only posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [ ] Well done on the tests you've so far. However, you need to have more tests coverage. Write unit tests for all your models and add more integration test for authentication and creating a post.

- [ ] Your friendship controller contains a lot of logic, let the model do the heavy lifting of managing business logic, controller should just manage the connection between the model and views. Use helper methods for Friendship. See this [article](https://smartfunnycool.com/friendships-in-activerecord/) for hint.

- [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (luisvinicius09) [reply app/views/layouts/application.html.erb:25]: You should update the scaffold then!

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔✔✔**

Hello @luisvinicius09 
Great work making the requested changes.
You can move on👍👌👌👌

**Pull request merged**